### PR TITLE
docs: EXPOSED-600 Add links to API docs for functions and classes

### PR DESCRIPTION
**Pull request by vnikolova**
#### Description

Add links to the API docs for functions and classes and fix their formatting where necessary.

#### Details

This PR focuses on the topics covered within the following sections:
- Databases
- Schemas
- Deep dive into DSL
- Deep dive into DAO

And applies the following changes:
1. Add a link to the API docs on the first mention of an Exposed function, class, or interface.
2. Format functions per the Ktor Style Guide.
3. Use descriptive titles instead of function names where appropriate. (mainly in the 'CRUD operations' topics)

[Preview on staging](https://writerside.labs.jb.gg/staging/exposed/vnikolova/EXPOSED-600/home.html)
---

#### Type of Change

- [x] Documentation update

---

#### Related Issues

[EXPOSED-600](https://youtrack.jetbrains.com/issue/EXPOSED-600) Add links to the API Documentation for functions and classes

### CODE REVIEW

File: documentation-website/Writerside/topics/DSL-Querying-data.topic [hunk 1]
```diff
@@ -269,30 +294,55 @@
...
         <chapter title="Group by" id="group-by">
-            <p>In <code>groupBy</code>, define fields and their functions by the <code>select()</code>
-                method.</p>
+            <p>
+                In
+                <a href="https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql/-query/group-by.html">
+                    <code>.groupBy()</code>
+                </a>
+                , define fields and their functions by the <code>select()</code>
+                method.
```
Comment (bog-walk) [inline documentation-website/Writerside/topics/DSL-Querying-data.topic:303]: .select()?

File: documentation-website/Writerside/topics/SQL-Functions.md [hunk 2]
```diff
@@ -88,9 +105,9 @@ val replacedName = CustomFunction<String?>("REPLACE", VarCharColumnType(), FooTa
 `CustomFunction` class accepts a function name as a first parameter and the resulting column type as second. After that, you can provide any amount of parameters separated by a comma.
 
 There are also shortcuts for string, long, and datetime functions:
-* `CustomStringFunction`
-* `CustomLongFunction`
-* `CustomDateTimeFunction`
```
Comment (bog-walk) [inline documentation-website/Writerside/topics/SQL-Functions.md:93]: Could `CustomFunction` above (line 105) also get an api link? Can't seem to find one anywhere.

File: documentation-website/Writerside/topics/Transactions.md [hunk 3]
```diff
@@ -316,16 +322,16 @@ transaction {
...
 `registerInterceptor()` and `unregisterInterceptor()` can be used to enable and disable a custom interceptor in a single
 transaction.
 
-To use a custom interceptor that acts on all transactions, extend the `GlobalStatementInterceptor` class instead.
-Exposed uses the Java SPI ServiceLoader to discover and load any implementations of this class. In this situation, a new
-file should be created in the *resources* folder
-named:
+To use a custom interceptor that acts on all transactions, extend the
+[`GlobalStatementInterceptor`](https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql.statements/-global-statement-interceptor/index.html)
+class instead. Exposed uses the Java SPI ServiceLoader to discover and load any implementations of this class. In this
```
Comment (bog-walk) [inline documentation-website/Writerside/topics/Transactions.md:333]: Could you please switch both usages of 'class' with 'interface', for GlobalStatementInterceptor? And 'extend' -> 'implement'.

**Review by bog-walk — APPROVED** [review]

File: documentation-website/Writerside/topics/DAO-Table-Types.topic [hunk 4]
```diff
@@ -73,9 +76,14 @@
...
-            Each component column will be available for CRUD operations either individually (as for any standard column)
+            <a href="https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.dao.id/-composite-id-table/index.html">
+                <code>CompositeIdTable</code>
+            </a>
+            and mark each composite column using
+            <a href="https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql/-table/entity-id.html">
+                <code>.entityId()</code>
+            </a>
+            . Each component column will be available for CRUD operations either individually (as for any standard column)
             or all together as part of the <code>id</code> column:
```
[the remark this answers is not in the document]
Comment (vnikolova) [reply documentation-website/Writerside/topics/DAO-Table-Types.topic:87]: I should have probably excluded this from the PR but it was a suggestion by one of our technical writers. I'll adjust the formatting of other mentions of ID too to be consistent.

**Comment by vnikolova** [comment]
Thanks for the comments @bog-walk 🙏 Fixed!

**Pull request merged**